services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -6536,6 +6536,37 @@ public boolean isLeaderControllerFor(String clusterName) {
     return model.isLeader();
   }
 
+  @Override
+  public InstanceRemovableStatuses getAggregatedHealthStatus(
+      String cluster,
+      List<String> instances,
+      List<String> toBeStoppedInstances) {
+    InstanceRemovableStatuses statuses = new InstanceRemovableStatuses();
+
+    // If current controller is not the leader, redirect with leader URL
+    if (!isLeaderControllerFor(cluster)) {
+      statuses.setRedirectUrl(getLeaderController(cluster).getUrl(false));
```

Comment:
Let's return the port that corresponds to the type of call that the controller received - if the call was `https`, we should return `https` response. If the call was `http`, we should return a `http` response